kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NMSettingsConverterTest.java

```diff
@@ -0,0 +1,641 @@
+		givenIP4Status("netIPv4StatusManagedLan");
+		givenDhcpStatus(false);
+		givenMapWith("net.interface." + this.netInterface + ".config.dhcpClient4.enabled", this.KuraDhcpStatus);
+		givenMapWith("net.interface." + this.netInterface + ".config.ip4.status", this.kuraIP4Status);
+		givenMapWith("net.interface." + this.netInterface + ".config.ip4.address", "192.168.0.12");
+		givenMapWith("net.interface." + this.netInterface + ".config.ip4.prefix", (short) 25);
+		givenMapWith("net.interface." + this.netInterface + ".config.ip4.dnsServers", "1.1.1.1");
+		givenMapWith("net.interface." + this.netInterface + ".config.ip4.gateway", "192.168.0.1");
+		givenNetworkPropsCreatedWithTheMap(this.internetNetworkPropertiesInstanciationMap);
+		
+		whenBuildSettingsIsRunWith(this.networkProperties, Optional.empty(), netInterface,
+				NMDeviceType.NM_DEVICE_TYPE_ETHERNET);
+		
+		thenNoExceptionsHaveBeenThrown();
+		thenResultingBuildAllMapContains("ipv6", "method", "disabled");
+		thenResultingBuildAllMapContains("ipv4", "method", "manual");
+		thenResultingBuildAllMapContains("ipv4", "address-data", buildAddressDataWith("192.168.0.12", new UInt32(25)));
+		thenResultingBuildAllMapContains("connection", "id", "kura-" + this.netInterface + "-connection");
+		thenResultingBuildAllMapContains("connection", "interface-name", this.netInterface);
+		thenResultingBuildAllMapContains("connection", "type", "802-3-ethernet");
+	}
+
+	@Test
+	public void buildSettingsShouldWorkWithExpectedInputsEthernetAndWan() {
+		givenNetInterface("eth0");
+		givenIP4Status("netIPv4StatusManagedWan");
+		givenDhcpStatus(false);
+		givenMapWith("net.interface." + this.netInterface + ".config.dhcpClient4.enabled", this.KuraDhcpStatus);
+		givenMapWith("net.interface." + this.netInterface + ".config.ip4.status", this.kuraIP4Status);
+		givenMapWith("net.interface." + this.netInterface + ".config.ip4.address", "192.168.0.12");
+		givenMapWith("net.interface." + this.netInterface + ".config.ip4.prefix", (short) 25);
+		givenMapWith("net.interface." + this.netInterface + ".config.ip4.dnsServers", "1.1.1.1");
+		givenMapWith("net.interface." + this.netInterface + ".config.ip4.gateway", "192.168.0.1");
+		givenNetworkPropsCreatedWithTheMap(this.internetNetworkPropertiesInstanciationMap);
+
+		whenBuildSettingsIsRunWith(this.networkProperties, Optional.empty(), netInterface,
+				NMDeviceType.NM_DEVICE_TYPE_ETHERNET);
+
+		thenNoExceptionsHaveBeenThrown();
+		thenResultingBuildAllMapContains("ipv6", "method", "disabled");
+		thenResultingBuildAllMapContains("ipv4", "method", "manual");
+		thenResultingBuildAllMapContains("ipv4", "address-data", buildAddressDataWith("192.168.0.12", new UInt32(25)));
+		thenResultingBuildAllMapContains("connection", "id", "kura-" + this.netInterface + "-connection");
+		thenResultingBuildAllMapContains("connection", "interface-name", this.netInterface);
+		thenResultingBuildAllMapContains("connection", "type", "802-3-ethernet");
+	}
+
+	// given
```

Comment:
Use the already established format:

```suggestion
     /*
     * Given
     */
```